frappe/website/doctype/personal_data_download_request/test_personal_data_download_request.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies and Contributors
+# See license.txt
+from __future__ import unicode_literals
+
+import frappe
+import unittest
+from frappe.website.doctype.personal_data_download_request.personal_data_download_request import get_user_data
+
+
+class TestRequestPersonalData(unittest.TestCase):
+	def setUp(self):
+		create_user_if_not_exists(email='test_privacy@example.com')
+
+	def test_user_data(self):
+		user_data = get_user_data('test_privacy@example.com')
+		expected_data = {'Contact': frappe.get_all('Contact', {'email_id':'test_privacy@example.com'},["*"])}
+		self.assertEqual({'Contact': user_data['Contact']}, expected_data)
+
+	def test_file_and_email_creation(self):
+		frappe.set_user('test_privacy@example.com')
+		download_request = frappe.get_doc({"doctype": 'Personal Data Download Request', 'user': 'test_privacy@example.com'})
+		download_request.save(ignore_permissions=True)
+		frappe.set_user('Administrator')
+
+		f = frappe.get_all('File',
```

Comment:
f for ??

Reply:
Same as above

Reply:
i have written this as per the contribution guidelines.
make sure you are following it